nuxeo-runtime/nuxeo-runtime-deploy/src/main/java/org/nuxeo/runtime/deployment/preprocessor/install/commands/AppendCommand.java

```diff
@@ -94,4 +97,23 @@ public String toString(CommandContext ctx) {
         return "append " + ctx.expandVars(src.toString()) + " > " + ctx.expandVars(dst.toString());
     }
 
+    private void append(File srcFile, File dstFile, boolean appendNewLine) throws IOException {
+        String srcExt = FileUtils.getFileExtension(srcFile.getName());
+        String dstExt = FileUtils.getFileExtension(dstFile.getName());
+        if (!StringUtils.equalsIgnoreCase(srcExt, dstExt)) {
+            throw new IOException("Source and destination files have different extensions.");
```

Comment:
Thought it would be a good safeguard since appending text to a json will break things